lib/src/rules/prefer_final_locals.dart

```diff
@@ -100,3 +200,23 @@ class _Visitor extends SimpleAstVisitor<void> {
     }
   }
 }
+
+extension on Token? {
+  bool get isFinal {
+    var self = this;
+    if (self == null) return false;
+    return self.keyword == Keyword.FINAL;
+  }
+}
+
+extension on DartPattern {
+  DeclaredVariablePattern? get declaredVariablePattern {
```

Comment:
I don't understand the semantics you're trying to implement here.

Reply:
Removed in https://github.com/dart-lang/linter/pull/4225